python/lsst/afw/cameraGeom/assembleImage.py

```diff
@@ -129,9 +128,8 @@ def makeUpdatedDetector(ccd):
     ccd : `lsst.afw.image.Detector`
         The detector to copy and update.
     """
-    ampInfoCatalog = ccd.getAmpInfoCatalog().copy(deep=True)
-
-    for amp in ampInfoCatalog:
+    builder = ccd.rebuild()
+    for amp in builder.getAmplifiers():
         assert amp.getHasRawInfo()
 
         bbox = amp.getRawBBox()
```

Comment:
That almost makes it sound like what we really want are Python builders for C++ immutable objects 🤔. Though that would require at least a protected C++ constructor to get wrapped by Python, and since that would be an abuse of the access specifier system I can't actually recommend it. Let me see if I can find a better solution.